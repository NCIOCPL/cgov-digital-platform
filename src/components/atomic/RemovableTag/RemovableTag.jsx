import React from 'react';
import PropTypes from 'prop-types';
import './RemovableTag.scss';

const RemovableTag = ({
  label,
  onRemove
}) => {
  const handleClick = (e) => {
    console.log('remove');
    onRemove();
  }

  return (
    <div className="cts-removable-tag">
      <button className="cts-removable-tag__button" type="button" aria-label={`remove ${label}`} onClick={handleClick}><span aria-hidden="true">X</span></button>
      <span className="cts-removable-tag__label">{label}</span>
    </div>
  );
};

RemovableTag.propTypes = {
  label: PropTypes.string,
  onRemove: PropTypes.func
};

RemovableTag.defaultProps = {
  onRemove: () => {}
};

export default RemovableTag;
