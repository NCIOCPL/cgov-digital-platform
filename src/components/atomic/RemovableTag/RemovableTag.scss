.cts-removable-tag {
  display: inline-block;
  background-color: #2a71a5;
  border: 1px solid #2a71a5;
  margin: 5px 3px;
  height: 32px;
  border-radius: 16px;
  padding-left: 12px;
  padding-right: 6px;
  color: white;
  

  &__button {
    min-width: auto !important;
    min-height: auto;
    width: 18px;
    height: 18px;
    vertical-align: middle;
    border-radius: 50%;
    appearance: none;
    background: #003366;
    font-size: 11px;
    color: white;
    padding: 0;
    text-align: center;
    margin: 0 0 0 .75rem;
    cursor: pointer;
    &:hover,
    &:focus {
      color: #036;
      background: #fff;
      font-weight: bold;
    }
  }
  &__label {
    font-size: 14px;
    line-height: 28px;
  }
}
