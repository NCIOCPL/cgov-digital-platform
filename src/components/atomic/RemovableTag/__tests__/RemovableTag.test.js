import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import RemovableTag from '../RemovableTag';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <RemovableTag {...props} />
  );
  return component;
};

describe('RemovableTag', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
