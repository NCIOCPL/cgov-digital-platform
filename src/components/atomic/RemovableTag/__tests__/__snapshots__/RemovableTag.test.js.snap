// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemovableTag Render renders without error 1`] = `
<div
  aria-selected="true"
  class="cts-removable-tag"
  role="option"
>
  <span
    class="cts-removable-tag__label"
  />
  <button
    aria-label="remove undefined"
    class="cts-removable-tag__button"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      X
    </span>
  </button>
</div>
`;
